File: 0001 - 0300/0100. Same Tree/fang.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var preOrderTraverse = function (root) {
    let order = '';
    const s = [];
    let p = root;
    while (p || s.length) {
        if (p) {
            order += p.val;
            s.push(p);
            p = p.left;
        } else {
            order += '*';
            p = s.pop();
            p = p.right;
        }
    }

    return order;
}

var isSameTree = function (p, q) {
    return preOrderTraverse(p) == preOrderTraverse(q);
};

File: 1201 - 1500/1302. Deepest Leaves Sum/fang.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        stack<pair<TreeNode*, int>> s;
        TreeNode* p = root;
        int depth = 1;
        int maxDepth = 0;
        int result = 0;
        while (p || !s.empty()) {
            while (p) {
                s.push(make_pair(p, depth));
                p = p->left;
                depth++;
            }
            tie(p, depth) = s.top();
            if (depth > maxDepth) {
                maxDepth = depth;
                result = p->val;
            } else if (depth == maxDepth) {
                result += p->val;
            }
            p = p->right;
            s.pop();
            depth++;
        }
        return result;
    }
};